### PR TITLE
v2.11.0: Support newer C++ features on macOS-64 

### PR DESCRIPTION
**Pull request by tommy-waltmann**
Checklist
* [ ] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by tommy-waltmann** [comment]
@conda-forge-admin, please rerender

**Comment by tommy-waltmann** [comment]
@conda-forge-admin, please restart ci

**Comment by tommy-waltmann** [comment]
@conda-forge-admin, please rerender

File: recipe/build_osx_arm64.sh [hunk 1]
```diff
@@ -16,6 +16,12 @@ export CMAKE_PREFIX_PATH=${PREFIX}
 # Filter CMAKE_PREFIX_PATH out of CMAKE_ARGS because scikit-build needs to set it
 export SKBUILD_CONFIGURE_OPTIONS=$(echo $CMAKE_ARGS | sed -e 's/\-DCMAKE_INSTALL_PREFIX\=[^ ]* //g')
 
+# 10.14 required for some C++17 features used in TBB
```
Comment (bdice) [inline recipe/build_osx_arm64.sh:19]: Is this right? I thought we moved to C++17 for `std::cyl_bessel_j`, not TBB.

File: recipe/build_osx_arm64.sh [hunk 2]
```diff
@@ -16,6 +16,12 @@ export CMAKE_PREFIX_PATH=${PREFIX}
...
+MACOSX_DEPLOYMENT_TARGET:  # [osx and x86_64]
+  - "10.14"                # [osx and x86_64]
+MACOSX_SDK_VERSION:        # [osx and x86_64]
+  - "10.14"                # [osx and x86_64]
```
Comment (bdice) [inline recipe/build_osx_arm64.sh:23]: This isn't shell script. Does this go in `recipe/conda_build_config.yml`?

https://conda-forge.org/docs/maintainer/knowledge_base.html#requiring-newer-macos-sdks

Comment (tommy-waltmann) [reply recipe/build_osx_arm64.sh:23]: that's my fault, I'll fix it

File: recipe/build_osx_arm64.sh (hunk 1 above)
Comment (tommy-waltmann) [reply recipe/build_osx_arm64.sh:19]: We originally did it only for `std::cyl_bessel_j`, yes, but once we raised the C++ standard, there were some internal parts of tbb trying to use the newer standard and required an update to the target platform version to support it.

Comment (bdice) [reply recipe/build_osx_arm64.sh]: Nice to know! Thanks.

**Comment by tommy-waltmann** [comment]
@conda-forge-admin, please rerender

**Pull request merged**